ahnlich/ai/src/engine/ai/providers/providers.rs

```diff
@@ -0,0 +1,25 @@
+use std::path::PathBuf;
+use ahnlich_types::ai::AIModel;
+use crate::engine::ai::providers::fastembed::FastEmbedProvider;
+
+
+pub enum ModelProviders {
```

Comment:
Perhaps this is a binary argument i.e the model provider you want to use for download is `fast-embed`.... we can also set the default to fast-embed.... else if it is actually on a model-by-model use-case then perhaps we also add it to the `Model` enum within `Self::Text` and `Self::Image`?

Reply:
Something that may be more useful that we can keep in mind is the user providing their own models.